docs/content/policy-reference.md

```diff
@@ -74,10 +74,25 @@ complex types.
 | <span class="opa-keep-it-together">`output := object.union(objectA, objectB)`</span> | `output` is a new object which is the result of an asymmetric recursive union of two objects where conflicts are resolved by choosing the key from the right-hand object (`objectB`). For example: `object.union({"a": 1, "b": 2, "c": {"d": 3}}, {"a": 7, "c": {"d": 4, "e": 5}})` will result in `{"a": 7, "b": 2, "c": {"d": 4, "e": 5}}`  |
 | <span class="opa-keep-it-together">`filtered := object.filter(object, keys)`</span> | `filtered` is a new object with the remaining data from `object` with only keys specified in `keys` which is an array, object, or set of keys. For example: `object.filter({"a": {"b": "x", "c": "y"}, "d": "z"}, ["a"])` will result in `{"a": {"b": "x", "c": "y"}}`). |
 | <span class="opa-keep-it-together">`filtered := json.filter(object, paths)`</span> | `filtered` is the remaining data from `object` with only keys specified in `paths` which is an array or set of JSON string paths. For example: `json.filter({"a": {"b": "x", "c": "y"}}, ["a/b"])` will result in `{"a": {"b": "x"}}`). |
+| <span class="opa-keep-it-together">`output := json.remove(object, paths)`</span> | `output` is a new object which is the result of removing all keys specified in `paths` which is an array or set of JSON string paths. For example: `json.remove{"a": {"b": "x", "c": "y"}, ["a/b"]}` will result in `{"a": {"c": "y"}}`. | 
```

Comment:
NOTE: how do we deal with arrays nested in `object`?

Reply:
Not 100% sure I am going to answer what you're asking, but here goes 😄  We stick to the json patch spec for the path syntax (indexes are just numbers in the path string `a/1/b/c/0` etc) and for removal of array elements where we remove it and shift remainders left. The main difference with this and json patch is that they are not done sequentially so if someone had like:

```json
o := {"x": ["a", "b", "c"]}
```
And did:
```rego
json.remove(o, ["a/1", "a/1"])
```
You get
```
o == {"x": ["a", "c"]}
```

Rather than with json patch, if I understand correctly, you would get:

```
o == {"x": ["a"]}
```

Since the second "a/1" path would point at `"c"` at the time it is evaluated (after `"b"` was removed and the array remainder shifted left)

Reply:
I'll add a comment that explains the paths are not processed sequentially.. that might be an important detail for users of this.

Reply:
Thanks for clarifying!